src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -35,13 +44,11 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
 
             GeoServerResourceLoader loader =
                     GeoServerExtensions.bean(GeoServerResourceLoader.class);
-            File f =
-                    Resources.find(
-                            Resources.fromURL(Files.asResource(loader.getBaseDirectory()), path),
-                            true);
+            Resource resource = loader.get(Paths.convert(path));
+
             URL url = null;
-            if (f != null && f.exists()) {
-                url = f.toURI().toURL();
+            if (resource.getType() == Resource.Type.RESOURCE) {
+                url = URLs.fileToUrl(resource.file());
             } else {
                 url = new URL(path);
```

Comment:
I think the else clause may cause trouble. In any case, for the sake of clarity I'd suggest to return `new URL(path)` only of `resource.getType() == Resource.Type.UNDEFINED` meaning the file doesn't exist, and throw an exception otherwise (meaning there's a directory with that name, as the only other option is `Resource.Type.DIRECTORY`). In general when dealing with enums we shall cover all possibilities, usually with a `default` clause on a `switch` statement when we're only interested on a subset of the enum values?